docs-src/prod-readiness-context/data-encryption.md

```diff
@@ -0,0 +1,52 @@
+---
+id: data-encryption
+title: How to encrypt data using a custom Data Converter
+description: Configure your custom encryption logic with `PayloadCodec` and set it with a custom Data Converter.
+sidebar_label: Data Encryption
+tags:
+  - guide-context
+  - production-readiness
+---
+
+Temporal Server stores and persists the data handled in your Workflow Execution.
+Encrypting this data to ensures that any sensitive application data is secure when handled by the Temporal Server.
+
+For example, if you have sensitive information passed in the following objects that are persisted in the Workflow Execution Event History, use encryption to secure it:
+
+- Inputs and outputs/results in your [Workflow](/concepts/what-is-a-workflow-execution), [Activity](/concepts/what-is-an-activity-execution), and [Child Workflow](/concepts/what-is-a-child-workflow-execution)
+- Inputs to your [Signal](/concepts/what-is-a-signal)
+- [Memo](/concepts/what-is-a-memo)
+- Results of [Local Activity](/concepts/what-is-a-local-activity), [Side Effects](/concepts/what-is-a-side-effect)
+- [Application errors and failures](/kb/failures).
+
+Using encryption ensures that your sensitive data exists unencrypted only on the Client and the Worker process that is executing the Workflows and Activities, on hosts that you control.
+
+To encrypt your data, configure your custom encryption logic with [`PayloadCodec`](/concepts/what-is-a-payload-codec) and set it with a [custom Data Converter](/concepts/what-is-a-custom-data-converter).
+
+A [`PayloadCodec`](/concepts/what-is-a-payload-codec) transforms your payloads (for example, by implementing compression and/or encryption and decryption) and is an optional step that happens between the wire and the [Payload Converter](/concepts/what-is-a-payload-converter):
+
+```bash
+User code <--> Payload Converter <--> Payload Codec <--> Wire <--> Temporal Server
+```
+
+A `PayloadCodec` implementation is applied with a custom Data Converter in your Client options.
+
+You can run your `PayloadCodec` with a [Codec Server](/concepts/what-is-a-codec-server) and use the Codec Server endpoints in Web UI and tctl to decode your encrypted payloads locally.
+For details, see [Decoding payloads on the Web UI and tctl](/concepts/what-is-remote-data-encoding#decoding-payloads-on-the-web-ui-and-tctl).
+
+However, if you plan to set up remote data encoding for your data, ensure that you consider all security implications of running encryption remotely before implementing it.
+
+In codec implementations, we recommend running the function (such as compression or encryption) on the entire input payload, and putting the result in the data field of a new payload with a different `encoding` metadata field.
+That way, the input payload's metadata is preserved, and when the encoded payload is sent to be decoded, you can verify the metadata field before applying the decryption.
```

Comment:
Yes; I need a bit of explanation on this though. Will check in with you and update accordingly.

Reply:
Added a note
`Note that if your data is not encoded, we recommend passing the unencoded data to the decode function instead of failing the conversion.`